src/InstrumentationEngine.Attach/InstrumentationConfigurationSources.xsd

```diff
@@ -0,0 +1,53 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
+    <xs:element name="InstrumentationConfigurationSources">
+        <xs:complexType>
+            <xs:sequence>
+                <xs:element name="InstrumentationConfigurationSource" maxOccurs="10">
+                    <xs:complexType>
```

Comment:
><xs:complexType> [](start = 20, length = 16)

Can you flatten this file? You can do this by moving all of the complexType declarations to be children of the schema element, name them (e.g. InstrumentationConfigurationSourceType for this one), and then refer to the type name on each element.

Reply:
Yes, I can refactor this file; this was the generated schema Visual Studio made from the xml file I provided

---
In reply to: [387831461](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r387831461) [](ancestors = 387831461)